src/Http/Http.Extensions/test/RequestDelegateGenerator/Baselines/MapGet_NoParam_StringReturn_WithFilter.generated.txt

```diff
@@ -0,0 +1,206 @@
+{
+    %GENERATEDCODEATTRIBUTE%
+    internal class SourceKey
+    {
+        public string Path { get; init; }
+        public int Line { get; init; }
+
+        public SourceKey(string path, int line)
+        {
+            Path = path;
+            Line = line;
+        }
+    }
+}
+
+// This class needs to be internal so that the compiled application
+// has access to the strongly-typed endpoint definitions that are
+// generated by the compiler so that they will be favored by
+// overload resolution and opt the runtime in to the code generated
+// implementation produced here.
+%GENERATEDCODEATTRIBUTE%
+internal static class GenerateRouteBuilderEndpoints
+{
+    private static readonly string[] GetVerb = new[] { HttpMethods.Get };
+    private static readonly string[] PostVerb = new[] { HttpMethods.Post };
+    private static readonly string[] PutVerb = new[]  { HttpMethods.Put };
+    private static readonly string[] DeleteVerb = new[] { HttpMethods.Delete };
+    private static readonly string[] PatchVerb = new[] { HttpMethods.Patch };
+
+    %GENERATEDCODEATTRIBUTE%
+internal static global::Microsoft.AspNetCore.Builder.RouteHandlerBuilder MapGet(
+        this global::Microsoft.AspNetCore.Routing.IEndpointRouteBuilder endpoints,
+        [global::System.Diagnostics.CodeAnalysis.StringSyntax("Route")] string pattern,
+        global::System.Func<string> handler,
+        [global::System.Runtime.CompilerServices.CallerFilePath] string filePath = "",
+        [global::System.Runtime.CompilerServices.CallerLineNumber]int lineNumber = 0)
+    {
+        return global::Microsoft.AspNetCore.Http.Generated.GeneratedRouteBuilderExtensionsCore.MapCore(endpoints, pattern, handler, GetVerb, filePath, lineNumber);
+    }
+
+}
+
+namespace Microsoft.AspNetCore.Http.Generated
+{
+    using System;
+    using System.Collections;
+    using System.Collections.Generic;
+    using System.Collections.ObjectModel;
+    using System.Diagnostics;
+    using System.Linq;
+    using System.Reflection;
+    using System.Threading.Tasks;
+    using System.IO;
+    using Microsoft.AspNetCore.Routing;
+    using Microsoft.AspNetCore.Routing.Patterns;
+    using Microsoft.AspNetCore.Builder;
+    using Microsoft.AspNetCore.Http;
+    using Microsoft.Extensions.DependencyInjection;
+    using Microsoft.Extensions.FileProviders;
+    using Microsoft.Extensions.Primitives;
+
+    using MetadataPopulator = System.Func<System.Reflection.MethodInfo, Microsoft.AspNetCore.Http.RequestDelegateFactoryOptions?, Microsoft.AspNetCore.Http.RequestDelegateMetadataResult>;
+    using RequestDelegateFactoryFunc = System.Func<System.Delegate, Microsoft.AspNetCore.Http.RequestDelegateFactoryOptions, Microsoft.AspNetCore.Http.RequestDelegateMetadataResult?, Microsoft.AspNetCore.Http.RequestDelegateResult>;
+
+    %GENERATEDCODEATTRIBUTE%
+    file static class GeneratedRouteBuilderExtensionsCore
+    {
+        internal static class GenericThunks<T>
```

Comment:
Should we even generate this class if it is empty? It kind of looks weird. It also looks like a bug when you look at the `MapCore<T>` method that indexes into the empty dictionary in this class.

Seems like more code to compile, that isn't used.

Reply:
Yea, this needs to be cleaned up.

Reply:
Switched to only generated these if we produced thunks from the codegen.